lib/ui/screens/search/search_view.dart

```diff
@@ -1,20 +1,65 @@
-import 'package:flutter/material.dart';
+import 'package:flutter/material.dart' hide SearchController;
 import 'package:fpdart/fpdart.dart';
+import 'package:medi_support/ui/screens/search/search_controller.dart';
+import 'package:medi_support/ui/screens/search/search_model.dart';
 import 'package:medi_support/ui/widgets/custom_app_bar.dart';
+import 'package:medi_support/ui/widgets/custom_search_bar.dart';
+import 'package:medi_support/ui/widgets/icon_row.dart';
 
 class SearchView extends StatelessWidget {
-  const SearchView({super.key});
+  const SearchView({
+    super.key,
+    required this.controller,
+    required this.model,
+  });
+  final SearchController controller;
+  final SearchModel model;
 
   @override
   Widget build(BuildContext context) => Scaffold(
         appBar: CustomAppBar(
           title: some('Search'),
           leading: some(
             IconButton(
-              onPressed: Navigator.of(context).pop,
+              onPressed: controller.goBack,
               icon: const Icon(Icons.arrow_back),
             ),
           ),
         ),
+        body: Column(
+          children: <Widget>[
+            CustomSearchBar(
+              onSearch: controller.onSearch,
+              currentQuery: model.query,
+            ),
+            ListView.builder(
+              shrinkWrap: true,
+              itemCount: model.filteredResults.length,
+              itemBuilder: (BuildContext context, int index) => ListTile(
+                title: Text(model.filteredResults[index]),
+              ),
+            ),
+            if (model.query.isEmpty) ...<Widget>[
+              IconRow(
+                iconSize: 24,
+                icons: const <IconWithLabel>[
+                  IconWithLabel(Icons.search, 'Search'),
+                  IconWithLabel(Icons.home, 'Home'),
+                  IconWithLabel(Icons.favorite, 'Favorites'),
+                ],
+                onIconPressed: controller.onSearch,
+              ),
+              IconRow(
```

Comment:
**suggestion (non-blocking):** Only use one row or change the categories

Reply:
dummy data